config/pmix_setup_hwloc.m4

```diff
@@ -52,72 +50,58 @@ AC_DEFUN([PMIX_SETUP_HWLOC],[
                              [hwloc_topology_init],
                              [],
                              [pmix_hwloc_support=0])],
-          [PMIX_FLAGS_APPEND_UNIQ([PMIX_DELAYED_LIBS], [$with_hwloc_extra_libs])])
+          [PMIX_FLAGS_APPEND_UNIQ([PMIX_FINAL_LIBS], [$with_hwloc_extra_libs])])
 
     if test $pmix_hwloc_support -eq 0; then
-        AC_MSG_WARN([PMIx requires HWLOC topology library support, but])
+        AC_MSG_WARN([PRRTE requires HWLOC topology library support, but])
         AC_MSG_WARN([an adequate version of that library was not found.])
         AC_MSG_WARN([Please reconfigure and point to a location where])
         AC_MSG_WARN([the HWLOC library can be found.])
         AC_MSG_ERROR([Cannot continue.])
     fi
 
-    # update global flags to test for HWLOC version
-    PMIX_FLAGS_PREPEND_UNIQ([CPPFLAGS], [$pmix_hwloc_CPPFLAGS])
-
-    AC_MSG_CHECKING([if hwloc version is 1.5 or greater])
-    AC_COMPILE_IFELSE(
-          [AC_LANG_PROGRAM([#include <hwloc.h>],
-          [[
-    #if HWLOC_API_VERSION < 0x00010500
-    #error "hwloc version is less than 0x00010500"
-    #endif
-          ]])],
-          [AC_MSG_RESULT([yes])],
-          [AC_MSG_RESULT([no])
-           AC_MSG_ERROR([Cannot continue])])
-
-    AC_MSG_CHECKING([if hwloc version is 1.8 or greater])
-    AC_COMPILE_IFELSE(
-          [AC_LANG_PROGRAM([#include <hwloc.h>],
-          [[
-    #if HWLOC_API_VERSION < 0x00010800
-    #error "hwloc version is less than 0x00010800"
-    #endif
-          ]])],
-          [AC_MSG_RESULT([yes])
-           pmix_have_topology_dup=1],
-          [AC_MSG_RESULT([no])])
+    # NOTE: We have already read PMIx's VERSION file, so we can use
+    # those values
+    pmix_hwloc_min_num_version=PMIX_HWLOC_NUMERIC_MIN_VERSION
+    pmix_hwloc_min_version=PMIX_HWLOC_MIN_VERSION
+    AC_MSG_CHECKING([version at or above v$pmix_hwloc_min_version])
+    AC_PREPROC_IFELSE([AC_LANG_PROGRAM([
+                                        #include <hwloc.h>
+                                        #if (HWLOC_API_VERSION < $pmix_hwloc_min_num_version)
```

Comment:
Actually, I just took a quick look: I think the problem is here.

The value that you're reading from `VERSION` (i.e., in `$pmix_hwloc_min_version`) is of the form "1.11.0".
But `HWLOC_API_VERSION` is a hex number, like `0x00020800`.

So you're comparing apples and oranges.

So I think you want to decide whether you want to compare hwloc's API version or package version.

If you want to compare the package version, you can do a trick just like we introduced into PRRTE to compare the PMIx package version number: for each of the A.B.C version numbers you want to compare, concatenate them and make them a big hex number.  Then you can simply compare the 2 hex numbers.

Like this: ...er, the change that was just merged in #1819 looks like it was subsequently removed. ☹️ 

If you look at the diff from #1819 in prte_setup_pmix.m4, new lines 84-89, I convert an A.B.C value to a large hex number that I can then directly compare to another hex number, effectively comparing A.B.C vs. D.E.F to see which one is larger.

Was that removed because it didn't work?

Reply:
No - I'm just trying to make the configure codes consistent. Would rather not have every dependent library with its own method for checking against min version. So I changed `autogen.pl` to generate the numbers (ported from OMPI) and verified that the hex numbers were being properly created. I then use those values in the configure code instead of re-generating them.

If you look in `config/autogen_found_items.m4`, you'll see the results - easy way to ensure that we are comparing apples-to-apples. Looking locally, `PMIX_HWLOC_NUMERIC_MIN_VERSION` is set to `0x00010b00`, which is less than the `0x00010b06` found in HWLOC 1.11.13.

I agree it's an API version - frustrating that HWLOC doesn't provide the actual numeric version. I'm thinking of revising the tests to use `HWLOC_VERSION_MAJOR` and friends as a workaround.

Reply:
Sadly, v1.11 doesn't have the major/minor/release numbers available - so all you can do is check the api version. Not ideal. I suppose we could have one method for checking pre-2.0 that uses the API version, and then another that looks at the real version for 2.0 and above - but that seems like a lot of work for little gain. 🤷‍♂️

Reply:
Think I might have figured out the problem. These failures are all cases where HWLOC is not located in a system location, and therefore must be specified by `--with-hwloc=path`. In the configure code, however, I didn't alter the `CPPFLAGS` to point to where the include file can be found. So I'm thinking that without that alteration, we are probably picking up some old include file that doesn't meet our requirement in the test.

I'll give that a try and see if it fixes things.